ocaml/xapi/xapi_mgmt_iface.ml

```diff
@@ -186,7 +186,12 @@ let wait_for_clustering_ip ~__context ~(self : API.ref_Cluster_host) =
   debug "Waiting for clustering IP on bridge %s (%s)" bridge (Ref.string_of pIF);
   let get_ip () = Helpers.get_primary_ip_addr ~__context bridge primary_address_type in
   let is_connected () = Helpers.get_bridge_is_connected ~__context bridge in
-  wait_for_ip get_ip is_connected
+  let value = wait_for_ip get_ip is_connected in
+  begin match primary_address_type with
+  | `IPv4 -> Db.PIF.set_IP ~__context ~self:pIF ~value
+  | `IPv6 -> Db.PIF.set_IPv6 ~__context ~self:pIF ~value:[value]
```

Comment:
This would remove any additional (IPv6) addresses that the interface may have (e.g. normally at least the link-local one).

Reply:
Good point, should I get + add instead?